crates/topos-tce-transport/src/lib.rs

```diff
@@ -5,8 +5,51 @@ use serde::{Deserialize, Serialize};
 use std::str::FromStr;
 use thiserror::Error;
 use topos_core::uci::{Certificate, CertificateId};
-use topos_crypto::messages::{Address, Signature, H160};
-use topos_p2p::PeerId;
+use topos_crypto::messages::Signature;
+use topos_crypto::messages::{Address, H160};
+
+#[derive(Debug, Error)]
+pub enum ValidatorIdConversionError {
+    #[error("Failed to parse address string as H160")]
+    ParseError,
+    #[error("Failed to convert byte array into H160")]
+    InvalidByteLength,
+}
+
+#[derive(Clone, Copy, Debug, Serialize, Deserialize, Eq, PartialEq, Hash)]
+pub struct ValidatorId(H160);
+
+impl ValidatorId {
+    pub fn as_bytes(&self) -> &[u8] {
+        self.0.as_bytes()
+    }
+
+    pub fn address(&self) -> Address {
+        self.0
+    }
+}
+
+impl From<H160> for ValidatorId {
+    fn from(address: H160) -> Self {
+        ValidatorId(address)
+    }
+}
+
+impl TryFrom<&str> for ValidatorId {
```

Comment:
Can be replace by [FromStr](https://doc.rust-lang.org/std/str/trait.FromStr.html) implementation

Reply:
The last PR explicitly wanted a `TryFrom` impl instead a `FromStr` :)

Reply:
Mmh no it was switching from `From` to `TryFrom`.
`FromStr` is returning a `Result` to catch errors.